### PR TITLE
rtnl: add Addr* integration tests

### PR DESCRIPTION
**Pull request by yarikk**
Context in issue #40.

### CODE REVIEW

**Comment by jsimonetti** [comment]
```
--- SKIP: TestLiveAddrAddDel (0.00s)
    addr_live_test.go:79: AddrAdd:  netlink receive: operation not permitted
```
This is apparently due to not running as root. I never had use for this up to now.
Can you also update `.travis.yml` to run the integration tests as root? `sudo -E go test -v -tags=integration ./...` ?

**Comment by yarikk** [comment]
Tests reveal first bugs: 
```
=== RUN   TestLiveAddrs
--- PASS: TestLiveAddrs (0.00s)
    addr_live_test.go:59: * entry 0: &net.IPNet{IP:net.IP{0x7f, 0x0, 0x0, 0x1}, Mask:net.IPMask(nil)}
    addr_live_test.go:59: * entry 1: &net.IPNet{IP:net.IP{0xa, 0x14, 0x6, 0x0}, Mask:net.IPMask(nil)}
    addr_live_test.go:59: * entry 2: &net.IPNet{IP:net.IP{0xac, 0x11, 0x0, 0x1}, Mask:net.IPMask(nil)}
=== RUN   TestLiveAddrAddDel
--- FAIL: TestLiveAddrAddDel (0.00s)
    addr_live_test.go:86: address reported as added but can't confirm
    addr_live_test.go:89: AddrDel:  address 127.0.0.99: address not found
```

**Comment by jsimonetti** [comment]
> Tests reveal first bugs:
> 
> ```
> === RUN   TestLiveAddrs
> --- PASS: TestLiveAddrs (0.00s)
>     addr_live_test.go:59: * entry 0: &net.IPNet{IP:net.IP{0x7f, 0x0, 0x0, 0x1}, Mask:net.IPMask(nil)}
>     addr_live_test.go:59: * entry 1: &net.IPNet{IP:net.IP{0xa, 0x14, 0x6, 0x0}, Mask:net.IPMask(nil)}
>     addr_live_test.go:59: * entry 2: &net.IPNet{IP:net.IP{0xac, 0x11, 0x0, 0x1}, Mask:net.IPMask(nil)}
> === RUN   TestLiveAddrAddDel
> --- FAIL: TestLiveAddrAddDel (0.00s)
>     addr_live_test.go:86: address reported as added but can't confirm
>     addr_live_test.go:89: AddrDel:  address 127.0.0.99: address not found
> ```

There is also a parallelism problem in the tests. We should make sure to run the integration tests with -parallel=1

**Comment by yarikk** [comment]
> There is also a parallelism problem in the tests. We should make sure to run the integration tests with -parallel=1

You think? It shouldn't be: each test is independent, and should not interfere with other tests, potentially running in parallel.

**Comment by jsimonetti** [comment]
If I add 127.0.0.99/32 to dev lo manually, I only see that address in the TestAddrs output when I have -parallel=1

**Comment by yarikk** [comment]
> If I add 127.0.0.99/32 to dev lo manually, I only see that address in the TestAddrs output when I have -parallel=1

That's fine, TestAddrs doesn't rely on its presence or absence.

**Comment by yarikk** [comment]
@jsimonetti @insomniacslk this is now ok - please review and merge.

**Comment by jsimonetti** [comment]
Lgtm!

File: rtnl/addr.go [hunk 1]
```diff
@@ -87,7 +90,7 @@ func (c *Conn) Addrs(ifc *net.Interface, family int) (out []*net.IPNet, err erro
 			iplen := len(m.Attributes.Address)
 			a := &net.IPNet{
 				IP:   m.Attributes.Address,
-				Mask: net.CIDRMask(int(m.PrefixLength), iplen),
+				Mask: net.CIDRMask(int(m.PrefixLength), iplen<<3),
```
Comment (insomniacslk) [inline rtnl/addr.go]: On mobile, I am not sure what this bitshift means here. Could you add a comment that explains it?

Comment (yarikk) [reply rtnl/addr.go]: You're right. Let me rephrase the code.

**Comment by insomniacslk** [comment]
waiting on @jsimonetti for merging
